### PR TITLE
IDEM-9

### PR DESCRIPTION
**Pull request by pondpiu**
[#9 ](https://github.com/dltxio/idem-mobile/issues/9)

- publish PGP Endpoint : save pgp and send email to user email to verify their email (link in email redirect user to idem website)
- verify endpoint : for idem website to call to verify user email

### CODE REVIEW

File: .env.local [hunk 1]
```diff
@@ -21,4 +21,8 @@ IDEM_URL="https://idem.com.au/"
 OTP_HASHING_SECRET="SUPER_SECRET_KEY_FOR_HASHING"
 MESSAGEBIRD_API_KEY="MESSAGEBIRD_API_KEY"
 DIGITALSURGE_SIGNUP_ENDPOINT="https://sita.digitalsurge.io/api/partner/createcustomer/"
-DIGITALSURGE_PARTNER_TOKEN="<DIGITALSURGE_PARTNER_TOKEN>"
\ No newline at end of file
+DIGITALSURGE_PARTNER_TOKEN="<DIGITALSURGE_PARTNER_TOKEN>"
+MAILJET_API_KEY="<MAILJET_API_KEY>"
+MAILJET_SECRET="<MAILJET_SECRET>"
+FROM_EMAIL_ADDRESS="support@dltx.io"
+PROXY_API_URL= "http://localhost:3000/user"
```
Comment (xiangzhe0812) [inline .env.local:28]: Need api url in .env.developement and env.production as well.

File: src/auth/auth-anonymous.guard.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { AuthGuard } from "@nestjs/passport";
```
Comment (xiangzhe0812) [inline src/auth/auth-anonymous.guard.ts:1]: Can we change the file name to auth-local-guard  somthing.

File: src/user/user.controller.ts [hunk 3]
```diff
@@ -204,4 +200,65 @@ export class UserController {
...
+    @ApiResponse({
+        status: HttpStatus.BAD_REQUEST
+    })
+    @Get("verify-email")
+    async verifyEmail(
+        @Query("email") email: string,
+        @Query("token") token: string,
+        @Res() res: Response
+    ): Promise<void> {
+        const isSuccess = await this.userService.verifyEmail(email, token);
```
Comment (xiangzhe0812) [inline src/user/user.controller.ts:250]: Do we need this return?

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
1. Font doesnt look the same as the site
2. What is hpbs?
3. The webpage should be on our site, not the proxy
4. The ooops doesnt look good.

**Comment by xcjlo985** [comment]
As discussed with @pondpiu and @xiangzhe0812 the implementation will change. Probably new PRs

File: src/interfaces.ts [hunk 4]
```diff
@@ -64,7 +65,14 @@ export enum ConfigSettings {
...
     HTTPS_PROXY_PASSWORD = "HTTPS_PROXY_PASSWORD",
-    HTTPS_PROXY_USERNAME = "HTTPS_PROXY_USERNAME"
+    HTTPS_PROXY_USERNAME = "HTTPS_PROXY_USERNAME",
+    MAILJET_API_KEY = "MAILJET_API_KEY",
+    MAILJET_SECRET = "MAILJET_SECRET",
+    FROM_EMAIL_ADDRESS = "FROM_EMAIL_ADDRESS",
+    PROXY_API_URL = "PROXY_API_URL",
+    WEBSITE_URL = "WEBSITE_URL",
+    JWT_SECRET = "JWT_SECRET",
+    JWT_EXPIRATION_TIME = "JWT_EXPIRATION_TIME"
```
Comment (xcjlo985) [inline src/interfaces.ts:75]: Replace TIME with the unit

**Review by xcjlo985 — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Review by xcjlo985 — APPROVED** [review]

File: src/user/user.service.ts [hunk 5]
```diff
@@ -284,4 +305,84 @@ export class UserService {
...
+            this.logger.error(error);
+            return false;
+        }
+    }
+
+    public async verifyEmail(email: string, token: string): Promise<boolean> {
+        try {
+            const user = await this.userRepository.findOneBy({
+                email: hashMessage(email)
+            });
```
Comment (bitcoinbrisbane) [inline src/user/user.service.ts:354]: Should we trim and to lower the email?

File: .env.development [hunk 6]
```diff
@@ -1,18 +1,14 @@
...
 IDEM_URL="https://idem.com.au/"
-OTP_HASHING_SECRET="SUPER_SECRET_KEY_FOR_HASHING"
-MESSAGEBIRD_API_KEY="MESSAGEBIRD_API_KEY"
 DIGITALSURGE_SIGNUP_ENDPOINT="https://sita.digitalsurge.io/api/partner/createcustomer/"
-DIGITALSURGE_PARTNER_TOKEN="<DIGITALSURGE_PARTNER_TOKEN>"
\ No newline at end of file
+FROM_EMAIL_ADDRESS="support@dltx.io"
+PROXY_API_URL= "https://uat-proxy.idem.com.au/user"
+WEBSITE_URL="https://www.idem.com.au/"
+JWT_EXPIRATION_SECONDS="21600"
```
Comment (bitcoinbrisbane) [inline .env.development:14]: do we need quotes here?
```suggestion
JWT_EXPIRATION_SECONDS=21600
```

File: .env.local [hunk 7]
```diff
@@ -21,4 +21,11 @@ IDEM_URL="https://idem.com.au/"
...
+WEBSITE_URL="https://www.idem.com.au/"
+JWT_SECRET="myVerificationTokenSecret"
+JWT_EXPIRATION_SECONDS="21600"
```
Comment (bitcoinbrisbane) [inline .env.local:31]: And here

```suggestion
JWT_EXPIRATION_SECONDS=21600
```

**Review by xcjlo985 — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**